src/cgame/cg_beacon.cpp

```diff
@@ -570,7 +570,14 @@ static void DrawBeacon( cbeacon_t *b )
 		Vector2Set( screen[ 1 ], cgs.bc.hudRect[1][0] - b->size/2,
 		                         cgs.bc.hudRect[1][1] - b->size/2 );
 		Vector2Subtract( b->pos, cgs.bc.hudCenter, b->clamp_dir );
-		ProjectPointOntoRectangleOutwards( b->pos, cgs.bc.hudCenter, b->clamp_dir, (const vec2_t*)screen );
+
+		// avoid division-by-zero
+		vec2_t clamp_dir;
```

Comment:
I don't know how to do it. 😅️

Anyway this is receiving a `vec2_t` and passing a `vec2_t`, so unless we change all the functions, maybe it's better to avoid a conversion right there?

Reply:
something like (I'll implement a VEC2GLM function compatible with vec2_t later):

```
glm::vec3 clamp_dir( b->clamp_dir[0], b->clamp_dir[1], b->clamp_dir[2] );
clamp_dir[0] = clamp_dir[0] ? clamp_dir[0] : 0.0001f;
clamp_dir[1] = clamp_dir[1] ? clamp_dir[1] : 0.0001f;

ProjectPointOntoRectangleOutwards( b->pos, cgs.bc.hudCenter, &clamp_dir[0], (const vec2_t*)screen );
```

> maybe it's better to avoid a conversion right there?

Would not change anything. In theory, you could `reinterpret_cast<glm::vec2>( something )` and it would work.

Reply:
That workaround was totally removed as the division only occurred when the code should not run. The proper fix was then to avoid running the code when it shouldn't run.